internal/pkg/runtime/launcher/options.go

```diff
@@ -490,3 +493,11 @@ func OptSysContext(sc *types.SystemContext) Option {
 		return nil
 	}
 }
+
+// OptDevice sets CDI device mappings to apply.
+func OptDevice(op []string) Option {
+	return func(lo *Options) error {
+		lo.Device = op
```

Comment:
Does it make sense to call `IsQualifiedName` for each of the elements of `op` here to validate the options. Then it should not be required to perform this check explicitly later and the check for further validity can be delegated to the registry.

Reply:
Not really... this could be open to non-CDI `--device` usage in future, and as there's an intention to move the `Opt*` and stuff out of `internal`, for public consumption... I'd rather keep CDI specific stuff separate at present.